asmcomp/amd64/selection.ml

```diff
@@ -88,7 +88,7 @@ let pseudoregs_for_operation op arg res =
   (* One-address unary operations: arg.(0) and res.(0) must be the same *)
   | Iintop_imm((Iadd|Isub|Imul|Iand|Ior|Ixor|Ilsl|Ilsr|Iasr), _)
   | Iabsf | Inegf
-  | Ispecific(Ibswap (32|64)) ->
+  | Ispecific(Ibswap (32|64)|Isextend) ->
```

Comment:
The movsxd would support different source and target registers, no?

Reply:
Yes, I've lifted this restriction in the latest commit.

Reply:
If there a risk that the restriction was actually useful, in that it allowed to use more often the RAX case, resulting in better performance?

Reply:
I don't think so, the gains coming from the use of a single `cdqe` are often shadowed by the gains brought by generating less `mov` in `insert_op_debug` (the worst case is 2 `mov reg,reg` for a whopping 6 bytes)